dotNETStandard/HealthVaultProviderManagementPortal/HealthVaultProviderManagementPortal/Controllers/PatientController.cs

```diff
@@ -104,21 +104,64 @@ private static IList<TimelineEntryViewModel> ConvertToTimelineEntryViewModels(Ti
                 {
                     foreach (var schedule in snapshot.Schedules)
                     {
-                        timelineEntries.Add(new TimelineEntryViewModel
+                        if (schedule.LocalDateTime >= snapshot.EffectiveStartInstant.InUtc().LocalDateTime &&
+                            schedule.LocalDateTime <= snapshot.EffectiveEndInstant.InUtc().LocalDateTime)
                         {
-                            TaskId = task.TaskId,
-                            TaskName = task.TaskName,
-                            TaskImageUrl = task.TaskImageUrl,
-                            LocalDateTime = schedule.LocalDateTime,
-                            ScheduleType = schedule.Type
-                        });
+
+                            timelineEntries.Add(new TimelineEntryViewModel
+                            {
+                                TaskId = task.TaskId,
+                                TaskName = task.TaskName,
+                                TaskImageUrl = task.TaskImageUrl,
+                                LocalDateTime = schedule.LocalDateTime,
+                                ScheduleType = schedule.Type
+                            });
+
+                            if (schedule.Occurrences != null && schedule.Occurrences.Any(o => !o.InWindow))
+                            {
+                                var outOfWindowOccurrences = schedule.Occurrences.Where(o => !o.InWindow);
+                                foreach (var occurrence in outOfWindowOccurrences)
+                                {
+                                    AddOccurrence(timelineEntries, occurrence, task);
+                                }
+                            }
+                        }
+                    }
+                }
+
+                if (snapshot.UnscheduledOccurrences != null && snapshot.UnscheduledOccurrences.Any())
+                {
+                    foreach (var occurrence in snapshot.UnscheduledOccurrences)
+                    {
+                        {
+                            AddOccurrence(timelineEntries, occurrence, task);
+                        }
                     }
                 }
             }
 
             return timelineEntries;
         }
 
+        private static void AddOccurrence(List<TimelineEntryViewModel> timelineEntries, TimelineScheduleOccurrence occurrence, TimelineTask task)
```

Comment:
For transparency of what the method does, I would rename `AddOccurrence` to `AddUniqueOccurrence`.